### PR TITLE
Practice lab1 twitter data modeling

### PR DESCRIPTION
**Pull request by sofi-alvarado**

### CODE REVIEW

File: db/migrate/20230920200529_adding_reference.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddingReference < ActiveRecord::Migration[7.0]
+  def change
+
+  end
+end
```
Comment (ragar90) [inline db/migrate/20230920200529_adding_reference.rb:5]: If the migration was empty why did you created it?

File: db/migrate/20230920210641_adding_foreign_keys_to_follows.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class AddingForeignKeysToFollows < ActiveRecord::Migration[7.0]
+  def change
+    #Adding new columns for following and follower
+    add_column :follows, :following_user_id, :integer
+    add_column :follows, :follower_user_id, :integer
+    #Adding foreign keys to table follows making a reference with table users
+    add_foreign_key :follows, :users, column: :following_user_id
+    add_foreign_key :follows, :users, column: :follower_user_id
```
Comment (ragar90) [inline db/migrate/20230920210641_adding_foreign_keys_to_follows.rb:8]: All of this fields could have been added inside this migration why they were not included there?

File: db/migrate/20230920234645_add_fields_in_tweets_table.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class AddFieldsInTweetsTable < ActiveRecord::Migration[7.0]
+  def change
+    add_column :tweets, :retweet_id, :integer
+    add_column :tweets, :quoted, :boolean
+    add_column :tweets, :content, :text, limit: 280, null: false
+    add_column :tweets, :hashtags, :string, array: true
+
+    #Adding foreign key to table tweets making a reference with table users
+    #add_foreign_key :tweets, :users, column: :user_id
```
Comment (ragar90) [inline db/migrate/20230920234645_add_fields_in_tweets_table.rb:9]: Remove unnecessary comments

File: db/migrate/20230921011451_add_tweets_arrayt_at_hashtags.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddTweetsArraytAtHashtags < ActiveRecord::Migration[7.0]
+  def change
+    add_column :hashtags, :tweets, :string, array: true
```
Comment (ragar90) [inline db/migrate/20230921011451_add_tweets_arrayt_at_hashtags.rb:3]: Why are you defining this arrays field? it seems this option is only available for PostgreSQL, by adding it you are making your active record schema dependent of the database why are you adding this change?

File: app/models/tweet.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Tweet < ApplicationRecord
+  belongs_to :user
+  #self reference
+  belongs_to :retweet, class_name: "Tweet", optional: true
+  #a Tweet can have many replies
+  has_many :replies
+  #a bookmark belongs to a single Tweet
+  has_one :bookmarks
```
Comment (ragar90) [inline app/models/tweet.rb:8]: A tweet can be bookmarked by multiple users and at the same time a user can bookmark multiple tweets, `has_many` makes more sense for this releationship

**Review by ragar90 — APPROVED** [review]
Not bad in general, Just make sure to not create empty migrations and add comments to your database fields, especially if those can make your schema dependent of the RDBMS

File: db/migrate/20230920200529_adding_reference.rb (hunk 1 above)
Comment (sofi-alvarado) [reply db/migrate/20230920200529_adding_reference.rb:5]: It was an error, I know you have to create a new migration if you want to alter your database. But, in this particular case, I ran the command unintentionally.

File: db/migrate/20230920210641_adding_foreign_keys_to_follows.rb (hunk 2 above)
Comment (sofi-alvarado) [reply db/migrate/20230920210641_adding_foreign_keys_to_follows.rb:8]: To be honest, I realized I had to add those foreign keys just after I ran the db:migrate in the first migration. I know it doesn’t make sense to add a new empty migration and then creating another when I could simply add those in the first one, but it was a mistake I made😿

File: db/migrate/20230920234645_add_fields_in_tweets_table.rb (hunk 3 above)
Comment (sofi-alvarado) [reply db/migrate/20230920234645_add_fields_in_tweets_table.rb:9]: Got it!

File: db/migrate/20230921011451_add_tweets_arrayt_at_hashtags.rb (hunk 4 above)
Comment (sofi-alvarado) [reply db/migrate/20230921011451_add_tweets_arrayt_at_hashtags.rb:3]: I followed the basic structure you gave us for the table Hashtags in the ERD 🧐 I thought it was meant to store an array of hashtags, but I guess that's not how I should define it.

File: app/models/tweet.rb (hunk 5 above)
Comment (sofi-alvarado) [reply app/models/tweet.rb:8]: I didn’t take it in consideration. Thank you!

**Pull request merged**